### PR TITLE
Adding empty hostname verification

### PR DESCRIPTION
**Pull request by mexanick**
### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context

Required to prevent undefined behavior in case calling script is erroneus
## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: python/reg_interface_gem/scripts/gem_reg.py [hunk 1]
```diff
@@ -16,6 +16,9 @@
     if options.exe:
         parseXML()
         if (options.hostname):
+            if (options.hostname == ""):
+                print 'Please provide non-empty hostname'
+                exit
```
Comment (bdorney) [inline python/reg_interface_gem/scripts/gem_reg.py]: would change line 21 to `exit(os.EX_USAGE)` with an import on `os` somewhere.

File: python/reg_interface_gem/scripts/gem_reg.py [hunk 2]
```diff
@@ -16,9 +16,12 @@
     if options.exe:
         parseXML()
         if (options.hostname):
+            if (options.hostname == ""):
+                print 'Please provide non-empty hostname'
+                sys.exit(os.EX_USAGE)
             if (rpc_connect(options.hostname)):
```
Comment (jsturdy) [inline python/reg_interface_gem/scripts/gem_reg.py:22]: Would it not make more sense to do the check inside this method?
If it's not something we can change, we should ask UW

Comment (mexanick) [reply python/reg_interface_gem/scripts/gem_reg.py:22]: it's not something we want to change as the method comes from UW (I have a source code, but don't want to diverge). Also this doesn't go inline with quick fix. But in general - yes, I agree that it should simply throw the very same error as for bad hostname. I propose to merge this for the moment and request changes from UW.

**Review by bdorney — APPROVED** [review]

**Review by jsturdy — APPROVED** [review]
OK... my feelings on "quick fixes" are well known, but as this is the legacy branch anyway...

**Comment by bdorney** [comment]
Is there a reason this is unmerged? @mexanick

**Pull request merged**